src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1003,6 +1010,42 @@ public bool ProcessTransaction(Transaction transaction, int? blockHeight = null,
 
             lock (this.lockObject)
             {
+                // Do a pre-scan of the transaction inputs to see if they're used in other transactions already
+                foreach (TxIn input in transaction.Inputs)
+                {
+                    // See if this input is being used by another transaction.
+                    if (this.inputLookup.TryGetValue(input.PrevOut, out WalletIndexData indexData))
+                    {
+                        if (indexData.TransactionData.Id == hash)
+                            continue; // It's the same transaction, ignore.
+
+                        // Incoming transaction confirmed, existing transaction unconfirmed, delete existing
```

Comment:
Perhaps just add to this comment that `indexData` is data/transactions that is in our wallet.